app/controllers/api/emails_controller.rb

```diff
@@ -6,6 +6,18 @@ class Api::EmailsController < ApplicationController
   def create
     service = EmailToolSender.new(params[:page_id], email_params, tracking_params)
     if service.run
+      if service.action.is_a?(PendingActionService)
```

Comment:
@subbiahf22 I followed the same approach as here: https://github.com/SumOfUs/Champaign/blob/ea38a8dc10f5590cae7d08d7125e47c4a23fce7f/app/controllers/api/actions_controller.rb#L19-L30

I hope that works